### PR TITLE
CI: Fix docker build step in CreateRelease workflow

### PR DESCRIPTION
**Pull request by maxknv**
### Changelog category (leave one):
- Not for changelog (changelog entry is not required)

docker build to use docker file from a release commit

### Documentation entry for user-facing changes

- [ ] Documentation is written (mandatory for new features)

> Information about CI checks: https://clickhouse.com/docs/en/development/continuous-integration/

#### CI Settings (Only check the boxes if you know what you are doing):
- [ ]  Allow: All Required Checks
- [ ]  Allow: Stateless tests
- [ ]  Allow: Stateful tests
- [ ]  Allow: Integration Tests
- [ ]  Allow: Performance tests
- [ ]  Allow: All Builds
- [ ]  Allow: batch 1, 2 for multi-batch jobs
- [ ]  Allow: batch 3, 4, 5, 6 for multi-batch jobs
---
- [ ]  Exclude: Style check
- [ ]  Exclude: Fast test
- [ ]  Exclude: All with ASAN
- [ ]  Exclude: All with TSAN, MSAN, UBSAN, Coverage
- [ ]  Exclude: All with aarch64, release, debug
---
- [ ]  Run only fuzzers related jobs (libFuzzer fuzzers, AST fuzzers, etc.)
- [ ]  Exclude: AST fuzzers
---
- [ ]  Do not test
- [ ]  Woolen Wolfdog
- [ ]  Upload binaries for special builds
- [x]  Disable merge-commit
- [ ]  Disable CI cache

### CODE REVIEW

**Comment by robot-ch-test-poll1** [comment]
*This is an automated comment for commit 4f3a1657e32d5f7ee0c6a58a3c3eb56c9811293b with description of existing statuses. It's updated for the latest CI running*

[✅ Click here](https://s3.amazonaws.com/clickhouse-test-reports/72077/4f3a1657e32d5f7ee0c6a58a3c3eb56c9811293b/ci_running.html) to open a full report in a separate page

<details><summary>Successful checks</summary>
<table>
<thead><tr><th>Check name</th><th>Description</th><th>Status</th></tr></thead>
<tbody>
<tr><td>Docs check</td><td>Builds and tests the documentation</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/0f4990d2e79870970834fd30ce963d49ab502e73/docs_check.html">✅ success</a></td></tr>
<tr><td>Fast test</td><td>Normally this is the first check that is ran for a PR. It builds ClickHouse and runs most of <a href="https://clickhouse.com/docs/en/development/tests#functional-tests">stateless functional tests</a>, omitting some. If it fails, further checks are not started until it is fixed. Look at the report to see which tests fail, then reproduce the failure locally as described <a href="https://clickhouse.com/docs/en/development/tests#functional-test-locally">here</a></td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/05bd7f4b4cc2b087870877e9181e3c13539af723/fast_test.html">✅ success</a></td></tr>
<tr><td>Unit tests</td><td>Runs the unit tests for different release types</td><td><a href="https://s3.amazonaws.com/clickhouse-test-reports/0/05bd7f4b4cc2b087870877e9181e3c13539af723/unit_tests__release_.html">✅ success</a></td></tr>
<tbody>
</table>
</details>

File: .github/workflows/create_release.yml [hunk 1]
```diff
@@ -45,6 +50,11 @@ env:
         required: false
         default: false
         type: boolean
+      only-docker:
```
Comment (Felixoid) [inline .github/workflows/create_release.yml:53]: How terrible that we need so many duplicates here and there https://github.com/actions/runner/issues/1182#issue-935195181

File: tests/ci/ci_utils.py [hunk 2]
```diff
@@ -215,6 +215,20 @@ def is_latest_release_branch(branch):
         )
         return latest_branch == branch
 
+    @staticmethod
+    def is_latest_branch_per_first_version_segment(branch):
+        release_branches = Shell.get_output(
+            'gh pr list --label release --repo ClickHouse/ClickHouse --search "sort:created" --json headRefName'
```
Comment (Felixoid) [inline tests/ci/ci_utils.py:221]: It belongs to `version_helper`. There are [get_tagged_versions](https://github.com/ClickHouse/ClickHouse/blob/0ff7c280f9e032732b27f42bd5f5b5677e9fd723/tests/ci/version_helper.py#L353) and [get_supported_versions](https://github.com/ClickHouse/ClickHouse/blob/0ff7c280f9e032732b27f42bd5f5b5677e9fd723/tests/ci/version_helper.py#L364). It feels wrong to request the gh for this case.

**Review by Felixoid — COMMENTED** [review]
Besides the points, the workflow was already complex. Bash in yaml is generally unreadable, and with actions' templating it's unclear what's going on each step.

The way it goes scares me. Stepping into the yaml+bash from python looks like a downgrade.

The workflow so far looks like a working PoC. And this fix is rather an emergency. Then it means, we should plan rewriting it to the python's `contextmanager`.

File: tests/ci/ci_utils.py (hunk 2 above)
Comment (maxknv) [reply tests/ci/ci_utils.py:221]: it's an alternative which does not require full git history

Comment (Felixoid) [reply tests/ci/ci_utils.py:221]: but the whole workflow requires the full history. How do we set the tag to a commit otherwise?

**Review by Felixoid — APPROVED** [review]
It would be good to have a success running for the workflow before the merge. Let's do a few different runs:

- Deploy a historical docker for some stale tag, around 23.11 maybe?
- Deploy a new patch tag with dry-run
- And a complete deployment

**Pull request merged**